src/openrct2-ui/windows/Error.cpp

```diff
@@ -136,27 +136,42 @@ static void window_error_unknown5(rct_window* w)
  */
 static void window_error_paint(rct_window* w, rct_drawpixelinfo* dpi)
 {
-    int32_t t, l, r, b;
-
-    l = w->windowPos.x;
-    t = w->windowPos.y;
-    r = w->windowPos.x + w->width - 1;
-    b = w->windowPos.y + w->height - 1;
-
-    gfx_filter_rect(dpi, l + 1, t + 1, r - 1, b - 1, FilterPaletteID::Palette45);
-    gfx_filter_rect(dpi, l, t, r, b, FilterPaletteID::PaletteGlassSaturatedRed);
-
-    gfx_filter_rect(dpi, l, t + 2, l, b - 2, FilterPaletteID::PaletteDarken3);
-    gfx_filter_rect(dpi, r, t + 2, r, b - 2, FilterPaletteID::PaletteDarken3);
-    gfx_filter_rect(dpi, l + 2, b, r - 2, b, FilterPaletteID::PaletteDarken3);
-    gfx_filter_rect(dpi, l + 2, t, r - 2, t, FilterPaletteID::PaletteDarken3);
-
-    gfx_filter_rect(dpi, r + 1, t + 1, r + 1, t + 1, FilterPaletteID::PaletteDarken3);
-    gfx_filter_rect(dpi, r - 1, t + 1, r - 1, t + 1, FilterPaletteID::PaletteDarken3);
-    gfx_filter_rect(dpi, l + 1, b - 1, l + 1, b - 1, FilterPaletteID::PaletteDarken3);
-    gfx_filter_rect(dpi, r - 1, b - 1, r - 1, b - 1, FilterPaletteID::PaletteDarken3);
-
-    l = w->windowPos.x + (w->width + 1) / 2 - 1;
-    t = w->windowPos.y + 1;
-    draw_string_centred_raw(dpi, { l, t }, _window_error_num_lines, _window_error_text.data(), FontSpriteBase::MEDIUM);
+    ScreenCoordsXY leftTop{ w->windowPos.x, w->windowPos.y };
```

Comment:
Isn't `w->windowPos` already a ScreenCoordsXY? If so this and the following can be simplified
```suggestion
    ScreenCoordsXY leftTop{ w->windowPos };
```